src/Datadog.Trace.ClrProfiler.Managed/CallTarget/CallTargetState.cs

```diff
@@ -32,13 +35,29 @@ public CallTargetState(Scope scope, object state)
             _previousScope = null;
             _scope = scope;
             _state = state;
+            _startTime = null;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CallTargetState"/> struct.
+        /// </summary>
+        /// <param name="scope">Scope instance</param>
+        /// <param name="state">Object state instance</param>
+        /// <param name="startTime">The intended start time of the scope, intended for scopes created in the OnMethodEnd handler</param>
+        public CallTargetState(Scope scope, object state, DateTimeOffset? startTime)
+        {
+            _previousScope = null;
+            _scope = scope;
+            _state = state;
+            _startTime = startTime;
```

Comment:
Is there a reason why we don't chain the constructors?
Same comment for `CallTargetState(Scope, Scope, object)` below.

```suggestion
        public CallTargetState(Scope scope, object state, DateTimeOffset? startTime)
            : this(scope, state)
        {
            _startTime = startTime;
```

Reply:
My only reason for not chaining them is that struct's have to have all their fields initialized before being returned to the caller, so keeping the entire set of fields initializations in one method makes it pretty clear that we're doing this (even though the compiler would verify that of course).